smithy-linters/src/main/java/software/amazon/smithy/linters/MissingSensitiveTraitValidator.java

```diff
@@ -0,0 +1,233 @@
+
+/**
+ * <p>Validates that shapes and members that possibly contain sensitive data are marked with the sensitive trait.
+ */
+public final class MissingSensitiveTraitValidator extends AbstractValidator {
+    static final Set<String> DEFAULT_SENSITIVE_WORDS = SetUtils.of(
+            "authentication",
+            "authorization",
+            "bank",
+            "billing",
+            "birth",
+            "credential",
+            "email",
+            "ethnicity",
+            "insurance",
+            "license",
+            "passkey",
+            "passphrase",
+            "passport",
+            "password",
+            "phone",
+            "private",
+            "secret",
+            "sensitive",
+            "telephone",
+            "token",
+            "username",
+            "zip"
+    );
+    static final Set<String> DEFAULT_SENSITIVE_PHRASES = SetUtils.of(
+            "accesskey",
+            "accesstoken",
+            "auth",
+            "creditcard",
+            "firstname",
+            "lastname",
+            "plaintext",
+            "taxpayer"
+    );
+
+    public static final class Provider extends ValidatorService.Provider {
+        public Provider() {
+            super(MissingSensitiveTraitValidator.class, node -> {
+                NodeMapper mapper = new NodeMapper();
+                return new MissingSensitiveTraitValidator(
+                        mapper.deserialize(node, MissingSensitiveTraitValidator.Config.class));
+            });
+        }
+    }
+
+    /**
+     * MissingSensitiveTrait configuration.
+     */
+    public static final class Config {
+        private List<String> phrases = ListUtils.of();
+        private List<String> words = ListUtils.of();
+
+        private boolean excludeDefaults;
+
+        public List<String> getPhrases() {
+            return phrases;
+        }
+
+        public void setPhrases(List<String> phrases) {
+            this.phrases = phrases;
+        }
+
+        public List<String> getWords() {
+            return words;
+        }
+
+        public void setWords(List<String> words) {
+            this.words = words;
+        }
+
+        public boolean getExcludeDefaults() {
+            return excludeDefaults;
+        }
+
+        public void setExcludeDefaults(boolean excludeDefaults) {
+            this.excludeDefaults = excludeDefaults;
+        }
+    }
+
+    private final Set<String> sensitiveWords;
+    private final Set<String> sensitivePhrases;
+
+    private MissingSensitiveTraitValidator(Config config) {
+        if (!config.getExcludeDefaults()) {
+            Set<String> phrasesInit = new HashSet<>(DEFAULT_SENSITIVE_PHRASES);
+            Set<String> wordsInit = new HashSet<>(DEFAULT_SENSITIVE_WORDS);
+            phrasesInit.addAll(config.getPhrases()
+                    .stream()
+                    .map(phrase -> phrase.toLowerCase(Locale.US))
+                    .collect(Collectors.toSet()));
```

Comment:
you could just `.forEach(phrasesInit::add)` so you don't create an unnecessary intermediate set